src/basilisp/contrib/nrepl_server.lpy

```diff
@@ -139,8 +143,19 @@
               result  (last results)]
           (send-value request send-fn [result {:ns (str *ns*)}]))
         (catch python/Exception e
-          (info :eval-exception e)
-          (handle-error send-fn (assoc request :ns (str *ns*)) e))
+          (debug :eval-exception e)
+          (let [exc-type (condp instance? e
+                           basilisp.lang.reader/SyntaxError
+                           basilisp.lang.reader/SyntaxError
+
+                           basilisp.lang.compiler/CompilerException
+                           basilisp.lang.compiler/CompilerException
+
+                           python/Exception
+                           python/Exception)
+                msg (->> (basilisp.lang.exception/format_exception e exc-type (.-__traceback__ e))
```

Comment:
```suggestion
          (let [msg (->> (basilisp.lang.exception/format_exception e (type e) (.-__traceback__ e))
```

Does this suffice? I don't think the `condp` is really doing anything for us.

Reply:
Thanks! I used the basilisp.cli/repl function as a reference without much thought and tried to be clever by combining them under one handler 😅Should the REPL excp handler be simplified as well?

https://github.com/basilisp-lang/basilisp/blob/c6ce8a16af3bdc99778d4078b7ffb502665a6265/src/basilisp/cli.py#L454-L465

Reply:
Ah yeah. I suspect it probably can be collapsed to this:

```python
    except Exception as e:  # pylint: disable=broad-exception-caught 
        print_exception(e, type(e), e.__traceback__) 
        repl_module.mark_exception(e) 
        continue 
```

Or perhaps even this:

```python
    except Exception as e:  # pylint: disable=broad-exception-caught 
        print_exception(e) 
        repl_module.mark_exception(e) 
        continue 
```

Reply:
Should I piggyback this onto the current PR as well, or would you prefer to handle it separately?

Reply:
Maybe a separate PR (or don't worry about it -- it's not really broken, just more complicated than it needs to be).

Reply:
OK, I'll leave it as it is for now then, thanks